client/my-sites/stats/site.jsx

```diff
@@ -504,7 +495,16 @@ function StatsBody( { siteId, chartTab = 'views', date, context, isInternal, ...
 		setActiveLegend( period !== 'hour' ? newActiveTab.legendOptions || [] : [] );
 	}, [ chartTab, period, activeTabState, context.query ] );
 
-	const query = chartRangeToQuery( customChartRange );
```

Comment:
I noticed chartRangeToQuery was creating a new variable on each render which was causing the `query` variable to be recreated and ultimately many re-renders. Issue I noticed was this [causing many dispatches](https://github.com/Automattic/wp-calypso/blob/e93071be9020b4cf11eb9096b9faf89acf3c25bf/client/my-sites/stats/features/modules/stats-utm/stats-module-utm.jsx#L137) on the stats home screen because of the query of dependency in UTM stats

Reply:
This is not an unrelated change to this PR, though; this change is a blocker for this PR. If it is removed, it crashes the stats homepage because of too many dispatches during renders like I mentioned in original comment

![Arc -2025-05-08 at 08 32 16@2x](https://github.com/user-attachments/assets/6bd74413-e5d5-40c0-9049-cba8eadf55ae)